File: app/src/main/java/com/patientapp/health/PatientApp.java
package com.patientapp.health;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;
import android.os.Handler;
import android.os.Looper;

import com.google.firebase.analytics.FirebaseAnalytics;

public class PatientApp extends Application {

    public static final String CHANNEL_DAILY_FORM = "daily_health_form";

    @Override
    public void onCreate() {
        super.onCreate();
        createNotificationChannel();
        enableAnalyticsAfterStartup();
    }

    private void enableAnalyticsAfterStartup() {
        new Handler(Looper.getMainLooper()).postDelayed(() ->
                FirebaseAnalytics.getInstance(this).setAnalyticsCollectionEnabled(true), 3000);
    }

    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                    CHANNEL_DAILY_FORM,
                    "Daily Health Form Reminder",
                    NotificationManager.IMPORTANCE_HIGH);
            channel.setDescription("Reminder to fill out your daily health form");
            channel.enableVibration(true);
            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(channel);
        }
    }
}

File: app/src/main/java/com/patientapp/health/PatientApp.kt
package com.patientapp.health

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build
import com.google.firebase.analytics.FirebaseAnalytics

class PatientApp : Application() {
    override fun onCreate() {
        super.onCreate()
        createNotificationChannel()
        enableAnalyticsAfterStartup()
    }

    private fun enableAnalyticsAfterStartup() {
        android.os.Handler(mainLooper).postDelayed({
            FirebaseAnalytics.getInstance(this).setAnalyticsCollectionEnabled(true)
        }, 3000)
    }

    private fun createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_DAILY_FORM,
                "Daily Health Form Reminder",
                NotificationManager.IMPORTANCE_HIGH
            ).apply {
                description = "Reminder to fill out your daily health form"
                enableVibration(true)
            }
            val manager = getSystemService(NotificationManager::class.java)
            manager.createNotificationChannel(channel)
        }
    }

    companion object {
        const val CHANNEL_DAILY_FORM = "daily_health_form"
    }
}
